suite-native/module-authorize-device/src/screens/connect/ConnectBluetoothDeviceScreen.tsx

```diff
@@ -1,8 +1,48 @@
-import { BluetoothDeviceManager } from '@suite-native/bluetooth';
-import { Screen, ScreenHeader } from '@suite-native/navigation';
-
-export const ConnectBluetoothDeviceScreen = () => (
-    <Screen header={<ScreenHeader closeActionType="close" />}>
-        <BluetoothDeviceManager />
-    </Screen>
-);
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import {
+    BluetoothDeviceList,
+    selectNearbyBluetoothDevices,
+    selectUnknownNearbyBluetoothDevices,
+    useBluetoothDevice,
+} from '@suite-native/bluetooth';
+import {
+    AuthorizeDeviceStackParamList,
+    AuthorizeDeviceStackRoutes,
+    Screen,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+
+import { BluetoothDeviceScreenHeader } from '../../components/connect/BluetoothDeviceScreenHeader';
+
+type NavigationProps = StackNavigationProps<
+    AuthorizeDeviceStackParamList,
+    AuthorizeDeviceStackRoutes.ConnectBluetoothDevice
+>;
+
+export const ConnectBluetoothDeviceScreen = () => {
+    const { connectBluetoothDevice } = useBluetoothDevice();
+    const navigation = useNavigation<NavigationProps>();
+
+    const nearbyBluetoothDevices = useSelector(selectNearbyBluetoothDevices);
+    const unknownNearbyBluetoothDevices = useSelector(selectUnknownNearbyBluetoothDevices);
+
+    useEffect(() => {
+        if (nearbyBluetoothDevices.length === 0) {
+            navigation.goBack();
```

Comment:
note to myself: try this

Reply:
I personally don't like it (screens changing without user understanding why and what is happening), but it is designed like this, so let's see the feedback from others.